src/sources/util/unix.rs

```diff
@@ -0,0 +1,71 @@
+use crate::event::Event;
+use crate::sources::Source;
+use bytes::Bytes;
+use futures::{future, sync::mpsc, Future, Sink, Stream};
+use std::{fs::canonicalize, path::PathBuf};
+use tokio::{
+    self,
+    codec::{FramedRead, LinesCodec},
+};
+use tokio_uds::UnixListener;
+use tracing::field;
+use tracing_futures::Instrument;
+
+/**
+* A function that takes a single line of a received message and builds an Event object.
+* Used by build_unix_source to encode any logic necessary for transforming the raw message
+* into a usuable Event object (for example decoding syslog messages).
+**/
+type BuildEventFn = fn(&str, Option<Bytes>, &str) -> Option<Event>;
+
+/**
+* Returns a Source object corresponding to a Unix domain socket.  Passing in different functions
+* for build_event can allow for different source-specific logic (such as decoding syslog messages
+* in the syslog source).
+**/
+pub fn build_unix_source<'a>(
```

Comment:
I think this lifetime `<'a>` is extraneous here, as I am not seeing any references to it in the signature or body.

Reply:
Yeah sorry, that was left behind from when I was fighting lifetime rules here before.  Removed.